sw/host/opentitanlib/src/rescue/serial.rs

```diff
@@ -36,23 +41,52 @@ impl RescueSerial {
             uart,
             reset_delay: Duration::from_millis(50),
             enter_delay: Duration::from_secs(5),
+            version: Cell::default(),
         }
     }
 }
 
 impl Rescue for RescueSerial {
-    fn enter(&self, transport: &TransportWrapper, reset_target: bool) -> Result<()> {
-        log::info!("Setting serial break to trigger rescue mode.");
-        self.uart.set_break(true)?;
-        if reset_target {
-            transport.reset_target(self.reset_delay, /*clear_uart=*/ true)?;
+    fn enter(&self, transport: &TransportWrapper, mode: EntryMode) -> Result<()> {
+        log::info!("Setting serial break to trigger rescue mode (entry via {mode:?}).");
+        match mode {
+            EntryMode::Reset => {
+                self.uart.set_break(true)?;
+                transport.reset_target(self.reset_delay, /*clear_uart=*/ true)?;
+            }
+            EntryMode::Reboot => {
+                self.reboot()?;
+                self.uart.set_break(true)?;
+            }
+            EntryMode::None => {
+                self.uart.set_break(true)?;
+            }
         }
-        UartConsole::wait_for(&*self.uart, r"rescue:.*\r\n", self.enter_delay)?;
-        log::info!("Rescue triggered. clearing serial break.");
+        let version = UartConsole::wait_for(
+            &*self.uart,
+            r"rescue:(?:(\d+)\.(\d+))?.*\r\n",
+            self.enter_delay,
+        )?;
+        let version = if version.len() == 3 {
+            let major = version[1].parse::<u16>()?;
+            let minor = version[2].parse::<u16>()?;
+            (major << 8) | minor
+        } else {
+            0
+        };
+        self.version.set(version);
+        log::info!("Rescue triggered (version={version:04x}). clearing serial break.");
         self.uart.set_break(false)?;
         // Upon entry, rescue is going to tell us what mode it is.
         // Consume and discard.
         let _ = UartConsole::wait_for(&*self.uart, r"(ok|error):.*\r\n", Self::ONE_SECOND);
+
+        if version < Self::VERSION_1_0 {
```

Comment:
Could you please add a check that if `version >= 2.0`, then abort with an error that a newer opentitantool is required to communicate with this ROM_EXT.

Reply:
Done.